### PR TITLE
Implement improvements of Lua tests proposed at #1656

### PR DESCRIPTION
**Pull request by jgvictores**
This PR implements the proposal of #1656:
- Rename `test_string.lua` to more representative `test_port.lua`, enable it to  run without the `yarpserver`, and integrate it in the test infrastructure (and thus also removed from `bindings/lua/tests/run.sh`).

### CODE REVIEW

File: bindings/lua/tests/run.sh [hunk 1]
```diff
@@ -14,4 +14,4 @@ fi
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-lua $DIR/test_string.lua
+# lua $DIR/test_that_requires_yarpserver.lua
```
Comment (drdanz) [inline bindings/lua/tests/run.sh]: Is this a leftover?

Comment (drdanz) [reply bindings/lua/tests/run.sh]: Maybe add a comment, i.e.
```diff
-lua $DIR/test_string.lua
+# Any test that requires yarpserver would go here" 
+# lua $DIR/test_that_requires_yarpserver.lua

Comment (jgvictores) [reply bindings/lua/tests/run.sh]: Sure, added at 8b7aa10d720c17d08aa7b49fac45765f84a02fd9 : )

**Comment by drdanz** [comment]
Rebased and merged, thanks.

**Pull request merged**